lib/SlideParser/Parser.php

```diff
@@ -607,4 +630,61 @@ private function getImageHtml(array $match, string $mime, string $path): string
 
 		return $html;
 	}
+
+	public function handleAnnotations(Slide $slide): string
+	{
+		$matches = $this->match(self::ANNOTATION_PATTERN, $slide->content);
+
+		$updatedContent = $slide->content;
+
+		if (empty($matches)) {
+			return $updatedContent;
+		}
+
+		foreach ($matches as $match) {
+			$matchedTag = $match[0];
+			$referenceColorId = $match[1];
+			$annotationId = $match[2];
+			$referenceIconId = $match[3];
+			$referencedText = $match[4];
+
+			if (empty($annotationId)) return $slide->content;
+
+			if (empty(Annotation::find($annotationId))) {
+				\Log::warning("Slide with ID {$slide->id} has incorrect reference. Annotation with ID ${annotationId} does not exist");
+				$replacement = sprintf(self::REFERENCE_EMPTY_FORMAT, $referencedText);
+			} else {
+				$referenceData = [
+					'reference_source_id' => $slide->id,
+					'reference_source_type' => Slide::class,
+					'reference_target_id' => $annotationId,
+					'reference_target_type' => Annotation::class,
+				];
+
+				if (!empty($referenceIconId) && !empty(ReferenceIcon::find($referenceIconId))) {
+					$referenceData['reference_icon_id'] = $referenceIconId;
+				}
+
+				if (!empty($referenceColorId) && !empty(ReferenceColor::find($referenceColorId))) {
+					$referenceData['reference_color_id'] = $referenceColorId;
+				}
+
+				// This if is only to handle the legacy parser tags. In the past we didn't have two dimensions: colors and icons, only one.
+				// Legacy tags without declared icon and with color id equal to 4 should be handled as references without color but with image icon.
+				// This code can be removed once we know we won't import presentations from slides.com with legacy tags.
+				if ($referenceColorId == self::ANNOTATION_LEGACY_IMAGE_ID && empty($referenceIconId)) {
```

Comment:
```suggestion
				if ($referenceColorId === self::ANNOTATION_LEGACY_IMAGE_ID && empty($referenceIconId)) {
```

Reply:
Also, I'd extract it to a separate function.

Reply:
It's a string - it won't work. You have to cast to int before.

Reply:
Right, let's make it explicit then:
```suggestion
				if ((int)$referenceColorId === self::ANNOTATION_LEGACY_IMAGE_ID && empty($referenceIconId)) {
```